backend/fms_core/viewsets/sample_next_step_by_study.py

```diff
@@ -36,4 +44,51 @@ def destroy(self, request, pk=None):
         if removed:
             return Response(data={"details": removed}, status=status.HTTP_200_OK)
         else:
-            return HttpResponseBadRequest(errors or f"Missing sample-next-step-by-study ID to delete.")
\ No newline at end of file
+            return HttpResponseBadRequest(errors or f"Missing sample-next-step-by-study ID to delete.")
+
+   
+    @action(detail=False, methods=["get"])    
+    def summary_by_study(self, request):
+        """
+        Returns the number of samples queued at each step for either a single study or for all
+        studies. 
+        
+        The endpoint returns an array of objects. Each object contains a study ID and a list
+        of steps, where each step includes a count of the number of samples queued.
+
+        For each step, the step order ID, order, step name and count is returned.
+
+        The endpoint only returns steps that have at least one sample queued - steps
+        with zero samples are omitted from the results.
+
+        Args:
+        study__id__in: The study ID (optional)
```

Comment:
This is not really an Arg, i would add it as a note in the above description so the auto-doc does not include it as an argument in the docs.